packages/online-editor/src/authSessions/AuthSessionsContext.tsx

```
@@ -137,18 +143,29 @@ export function AuthSessionsContextProvider(props: PropsWithChildren<{}>) {
         const newAuthSessionStatus: [string, AuthSessionStatus][] = await Promise.all(
           [...(authSessions?.values() ?? [])].map(async (authSession) => {
             if (authSession.type === "git") {
-              const authProvider = authProviders.find(({ id }) => id === authSession.authProviderId);
+              const authProvider = authProviders.find(({ id }) => id === authSession.authProviderId) as GitAuthProvider;
```

Although we know this to be true, there's no hard guarantee on the type system that this will be the case. In this scenario, I prefer to throw an exception if `authProvider` is not a GitAuthProvider. WDYT?

---

Fixed by adding a new type guard and improving the current type checks.